src/data/repositories/D2DataElement.ts

```diff
@@ -0,0 +1,502 @@
+            });
+
+            const indicatorResponse = await this.d2Indicator.getByCodes([
+                ...actualTargetCodes,
+                ...benefitCodes,
+            ]);
+
+            const indicatorsByKeys = _.keyBy(indicatorResponse, indicator => indicator.code);
+
+            const dataElements = _(response.objects)
+                .map(d2DataElement => this.buildDataElement(d2DataElement, indicatorsByKeys))
+                .compact()
+                .value();
+
+            return dataElements;
+        });
+
+        return _(dataElementsImported).flatten().value();
+    }
+
+    async save(ids: Id[], dataElements: DataElement[], options: SaveOptions): Promise<object> {
+        const dataElementsImported = await promiseMap(_.chunk(ids, 100), async dataElementIds => {
+            const response = await this.api.models.dataElements
+                .get({
+                    fields: { $owner: true },
+                    filter: { id: { in: dataElementIds } },
+                    paging: false,
+                })
+                .getData();
+
+            const postDataElements = dataElementIds.map(
+                (dataElementId): PartialPersistedModel<D2ApiDataElement> => {
+                    const existingD2DataElement = response.objects.find(
+                        d2DataElement => d2DataElement.id === dataElementId
+                    );
+                    const dataElement = dataElements.find(
+                        dataElement => dataElement.id === dataElementId
+                    );
+                    if (!dataElement) {
+                        throw Error(`Cannot find dataElement ${dataElementId}`);
+                    }
+
+                    const defaultAggregationType: D2ApiDataElement["aggregationType"] = "SUM";
+                    const defaultDomainType: "AGGREGATE" | "TRACKER" = "AGGREGATE";
+                    const valueType: "INTEGER_ZERO_OR_POSITIVE" | "NUMBER" =
+                        dataElement.mainType.name === "benefit"
+                            ? "NUMBER"
+                            : "INTEGER_ZERO_OR_POSITIVE";
+
+                    return {
+                        ...(existingD2DataElement || {}),
+                        aggregationType: defaultAggregationType,
+                        attributeValues: this.buildAttributes(dataElement, this.config),
+                        categoryCombo: dataElement.disaggregation
+                            ? { id: dataElement.disaggregation.id }
+                            : undefined,
+                        code: dataElement.code,
+                        description: dataElement.description,
+                        domainType: defaultDomainType,
+                        formName: dataElement.formName,
+                        id: dataElement.id,
+                        name: dataElement.name,
+                        shortName: dataElement.shortName,
+                        valueType: valueType,
+                    };
```

Comment:
Comparing with https://platform-test.samaritanspurse.org/dhis-web-maintenance/index.html#/edit/dataElementSection/dataElement/GMi2DiuzQhf, I just saw one difference, the "store zero data values" flag (zeroIsSignificant). Let's set it always to true (project-monitoring-metadata/src/metadata.ts#615)